### PR TITLE
Feature/ui top panel hydrator++

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Replaces floats & widths in top panel with flexbox in both studio view + detailed view
- Adds tooltip for both name & description in both studio view + detailed view
- Removes max-length directive as it is no longer being used

This is the first set of changes for adding settings to top panel from bottom panel.

Bamboo Build: http://builds.cask.co/browse/CDAP-DRC3921-3

Things achieved out of these change,
- No max-length set on name & description in JS
- Automatic ellipsis on name & description (no js workaround) 
- Both name & description uses full width instead of having ellipsis even though we have extra space to the right.
- Makes things more responsive on smaller screens.

### CODE REVIEW

**Comment by elfenheart** [comment]
Seems like there are differences to our current top panel?

Current:
screen shot 2016-06-30 at 10 58 37 am

New:
screen shot 2016-06-30 at 10 59 53 am

**Comment by ajainarayanan** [comment]
Latest Bamboo build: http://builds.cask.co/browse/CDAP-DRC3921-4

File: cdap-ui/app/features/hydratorplusplus/toppanel.less [hunk 1]
```diff
@@ -292,170 +260,115 @@ body.theme-cdap {
...
-      div.all-runs {
-        div.metric {
-          &:first-child {
-            padding-right: 5px;
-            width: 60%;
+          h1,
+          p {
+            white-space: nowrap;
+            overflow: hidden;
+            text-overflow: ellipsis;
```
Comment (elfenheart) [inline cdap-ui/app/features/hydratorplusplus/toppanel.less:314]: we really should create a mixin or helper class for this 😂

**Comment by elfenheart** [comment]
👍  LGTM

**Pull request merged**